File: LeetCode/Easy/Remove Elements/Cpp/remove_element.cc
#include <iostream>
#include <vector>

typedef std::vector<int> vi;

int remove_element(vi v, int val) {
    int j = 0;
    for (int i=0; i<v.size(); i++) {
        if (v[i] != val) {
            v[j] = v[i];
            j++;
        }
    }
    return j;
}

int main() {
    std::cout<<remove_element(vi{3,2,2,3}, 3)<<"\n";
}

File: LeetCode/Easy/Remove Elements/Java/removeElement.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k=nums.length-1;
        int i=0;
        if (nums.length == 0) return 0;
        while (i<k) {
            if (nums[i] == val){
                nums[i] = nums[k];
                k--;
            }else {
                i++;
            }
        }
        return nums[k] == val ? k : k+1;
    }
}
